src/config/generalconf.cpp

```diff
@@ -399,6 +408,57 @@ void GeneralConf::initAutostart()
       m_autostart, &QCheckBox::clicked, this, &GeneralConf::autostartChanged);
 }
 
+void GeneralConf::initHideCursor()
+{
+    const auto initDisabledHideCursor = [&]() -> void {
+        auto* disabledLabel = new QLabel(
+          tr(
+            "Cursor hiding behavior cannot be set on this desktop environment"),
+          this);
+        disabledLabel->setStyleSheet(STYLESHEET_DISABLED);
+        m_hideCursor =
+          new QCheckBox(tr("Hide cursor in screenshots (disabled)"), this);
+        m_hideCursor->setEnabled(false);
+        m_hideCursor->setToolTip(tr(
+          "Cursor hiding behavior cannot be set on this desktop environment"));
+        m_scrollAreaLayout->addWidget(disabledLabel);
+        m_scrollAreaLayout->addWidget(m_hideCursor);
+        connect(m_hideCursor,
+                &QCheckBox::clicked,
+                this,
+                &GeneralConf::hideCursorChanged);
+    };
+
+    const auto initEnabledHideCursor = [&]() -> void {
+        m_hideCursor = new QCheckBox(tr("Hide cursor in screenshots"), this);
+        m_hideCursor->setEnabled(true);
+        m_hideCursor->setToolTip(tr("Don't include the cursor in screenshots"));
+        m_scrollAreaLayout->addWidget(m_hideCursor);
+        connect(m_hideCursor,
+                &QCheckBox::clicked,
+                this,
+                &GeneralConf::hideCursorChanged);
+    };
+
+#if defined(Q_OS_LINUX)
+    if (!m_info.waylandDetected()) {
+        initDisabledHideCursor();
+    } else {
+        switch (m_info.windowManager()) {
```

Comment:
It might better to "opt in" desktops where we KNOW this feature will work, than just to opt out KDE and GNOME. We have people using basically every esoteric DE there is.